lib/rasr_cache.py

```diff
@@ -808,7 +808,7 @@ def read_str(self, l, enc="ascii"):
         """
         a = array.array("b")
         a.fromfile(self.f, l)
-        return a.tostring().decode(enc)
+        return a.tobytes().decode(enc)
```

Comment:
based on the fact that here is a `.decode(enc)` following the `tostring()` call, I assume the result already was a byte-string and we do not need to change anything else?

Reply:
The thing is tostring is removed for Python >=3.9, which means calling it directly results in an error. I assume since we are gradually shifting to higher Python versions and Python <3.2 is hardly used, it might make sense to change? But we can of course also just require people using lower Python version when calling the rasr_cache, not a problem for me.

Reply:
Just for clarifying: tostring and tobytes are absolutely the same thing with just different names, see [here](https://docs.python.org/3/library/array.html#array.array.tobytes) I don't know why they did it

Reply:
> and we do not need to change anything else?

I meant "change additional code because instead of a str we now get a bytes object" (which is not the case). I am totally in favor of the change.